src/main/java/net/neoforged/neoforge/fluids/capability/templates/FluidTank.java

```diff
@@ -60,14 +61,16 @@ public int getFluidAmount() {
     }
 
     public FluidTank readFromNBT(HolderLookup.Provider lookupProvider, CompoundTag nbt) {
-        FluidStack fluid = FluidStack.parseOptional(lookupProvider, nbt);
-        setFluid(fluid);
+        if (nbt.contains("Fluid", Tag.TAG_COMPOUND)) {
```

Comment:
Are you sure that this `if` check is useful? It will default to an empty compound already.

Reply:
I mean yeah if no data exists at `Fluid` it defaults to a empty CompoundTag but if some other data is there, a ClassCastException will be thrown, having the if changes it so it only loads if data is there and it is a CompoundTag, chance of throwing an error.

But i can remove the 'if' if thats preferred

Reply:
Does `getCompound` not catch the CCE anyway? It's more idiomatic to not have the if, I just want to check that it won't cause crashes.

Reply:
It does catch the exception but immediately throws a `ReportedException`

```java
public CompoundTag getCompound(String p_128470_) {
    try {
        if (this.contains(p_128470_, 10)) {
            return (CompoundTag)this.tags.get(p_128470_);
        }
    } catch (ClassCastException classcastexception) {
        throw new ReportedException(this.createReport(p_128470_, TYPE, classcastexception));
    }

    return new CompoundTag();
}
```

But then again, the `try { }` does check contains itself with tag type `10` with is `Tag.TAG_COMPOUND`, so yeah i guess the if is not really needed